spec/haskell/src/SEL4/Object/ObjectType.lhs

```diff
@@ -133,7 +133,9 @@ When the last capability to an endpoint is deleted, any IPC operations currently
 > finaliseCap (ReplyCap { capReplyPtr = ptr }) final _ = do
 >     when final $ do
 >         tptrOpt <- getReplyTCB ptr
->         when (tptrOpt /= Nothing) $ replyClear ptr
+>         when (tptrOpt /= Nothing) $ do
+>             let tptr = fromJust tptrOpt
+>             replyClear ptr tptr
```

Comment:
If it's only used once, I'd probably inline the `let` unless there is some other reason (e.g. to line it up with something in C or abstract), i.e.

```suggestion
>             replyClear ptr (fromJust tptrOpt)
```
(and removing the `let` above)